service-base/src/main/java/org/eclipse/hono/service/registration/DeviceRegistryAmqpServerBase.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service.registration;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.healthchecks.HealthCheckHandler;
+import io.vertx.ext.healthchecks.Status;
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.connection.ConnectionFactory;
+import org.eclipse.hono.service.amqp.AmqpServiceBase;
+import org.eclipse.hono.service.amqp.Endpoint;
+import org.eclipse.hono.service.auth.AuthenticationConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.Objects;
+
+/**
+ * Spring component that serves as the default implementation of Hono's device registry.
```

Comment:
It doesn't seem to be a Spring `Component` (no annotation)
It also is not the default implementation of Hono's device registry, is it? It is an abstract base class that can be used to create an implementation based on AMQP though ...